src/arrayMethodJoin.js

```diff
@@ -4,8 +4,35 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    let newSaparator = separator;
+    const newArr = this;
+    const notAdd = [null, undefined];
+
+    if (newSaparator === null) {
+      newSaparator = 'null';
+    }
+
+    for (let i = 0; i < newArr.length; i++) {
+      const cell = newArr[i];
+
+      if (notAdd.includes(cell)) {
```

Comment:
```suggestion
      if (cell !== null && cell !== undefined) {
```
Не потрібно створювати масив із цими значеннями